TLM/TLM/Manager/Impl/LaneConnection/LaneConnectionSubManager.cs

```diff
@@ -392,8 +393,36 @@ public class LaneConnectionSubManager :
                 }
             }
 
+            var connections = GetLaneConnections(sourceLaneId, sourceStartNode);
+            if (verbose_) {
+                Log._Debug($"AddLaneConnection: {sourceLaneId}->{targetLaneId} at {nodeId} connections={connections.ToSTR()}");
+            }
+            if (connections != null) {
+                foreach (uint laneId in connections) {
+                    LaneEnd key = new(laneId, nodeId);
+                    if (deadEnd) {
+                        if (laneId != sourceLaneId) {
+                            // dead end lane connection cannot have other lane connections.
+                            if (verbose_) {
+                                Log._Debug($"making a dead end connection disconnecting {sourceLaneId}->{laneId} at {nodeId}");
+                            }
+                            connectionDataBase_.Disconnect(sourceLaneId, laneId, nodeId);
+                        }
+                    } else {
+                        if (laneId == sourceLaneId) {
+                            // if adding a new connection then remove the dead end connection.
+                            if (verbose_) {
+                                Log._Debug($"removing dead end connection for lane:{sourceLaneId} at node:{nodeId}");
+                            }
+                            connectionDataBase_.Disconnect(sourceLaneId, sourceLaneId, nodeId);
+                        }
+                    }
+                }
+            }
+
             connectionDataBase_.ConnectTo(sourceLaneId, targetLaneId, nodeId);
             Assert(AreLanesConnected(sourceLaneId, targetLaneId, sourceStartNode), $"AreLanesConnected({sourceLaneId}, {targetLaneId}, {sourceStartNode})");
+            AssertLane(sourceLaneId, sourceStartNode);
```

Comment:
won't this call GetLaneConnections(..) with the same params again even though this was called already?

Reply:
yes but now it is after a connection is made so the return value would be different.

Reply:
right